addons/survey/models/survey_user_input.py

```diff
@@ -784,20 +784,14 @@ def _get_answer_score_values(self, vals, compute_speed_score=True):
 
         if compute_speed_score and answer_score > 0:
             user_input = self.env['survey.user_input'].browse(user_input_id)
-            session_speed_rating = user_input.exists() and user_input.is_session_answer and user_input.survey_id.session_speed_rating
+            session_speed_rating = user_input.exists() and user_input.survey_id.session_speed_rating
+            session_speed_rating_timer = question.time_limit if question.time_limit > 0 else user_input.survey_id.session_speed_rating_timer
             if session_speed_rating:
-                max_score_delay = 2
-                time_limit = question.time_limit
+                time_limit = session_speed_rating_timer
                 now = fields.Datetime.now()
                 seconds_to_answer = (now - user_input.survey_id.session_question_start_time).total_seconds()
-                question_remaining_time = time_limit - seconds_to_answer
-                # if answered within the max_score_delay => leave score as is
-                if question_remaining_time < 0:  # if no time left
-                    answer_score /= 2
-                elif seconds_to_answer > max_score_delay:
-                    time_limit -= max_score_delay  # we remove the max_score_delay to have all possible values
-                    score_proportion = (time_limit - seconds_to_answer) / time_limit
-                    answer_score = (answer_score / 2) * (1 + score_proportion)
+                temp_score = ((time_limit - seconds_to_answer) / time_limit) * answer_score / 2
```

Comment:
this is linear indeed, I was thinking of adding few "extra" seconds when score does not change, but then again, 
1) I'm not sure how we are going to decide how many seconds can one wait for each question as some questions can be 1+1= while others can take few paragraphs,
2) if someone can read questions faster and thus gives answer faster then others during that grace period, they don't get any bonus rating
Plus Francois seemed reluctant to make the calculation process more complicated, so I just decided to leave this part as is. 
Thank again, we can discuss it with Francois again once he comes back from vacation and try to think of better solution :D

Reply:
Agreed, if ok with François then ok with me ;) There could be a small field to store the number of seconds before losing points with default value of 0. Could be worth asking when he comes back ^^

Reply:
Actually this is insane: at the moment, as mentioned in the pad, if one uses 'reward quick answers', it is always 50% score ? Could be worth a 'fix' in stable like having a hard coded value of 30 sec or so in this function :p Worth asking frgi as well I think. (could be a separate task afaik)

Could you add a small test to assess this function computes correct scores when the option is used? You can look at appointment tests if you want, they use freeze time in order to avoid issues with '.now()' It is not too complicated :) 

You can look for `with freeze_time(self.reference_now) ...`

Reply:
Shouldn't we give a free seconds to answers ? Like answering in the {1...3} first seconds should probably be considered as getting all points, otherwise it is impossible to have them ?